### PR TITLE
Merge local check definition for extensions

### PR DESCRIPTION
**Pull request by Igorshp**
Alternative to https://github.com/sensu/sensu/pull/902

Looking for comments on how to make this cleaner, ideally removing duplication in lines 214 and 227.

### CODE REVIEW

**Comment by Igorshp** [comment]
Test failed, but it doesn't look related to this PR.... strange

**Comment by Igorshp** [comment]
retest

**Comment by Igorshp** [comment]
closing and reopenning PR to trigger travis build

**Comment by rmc3** [comment]
:+1:

**Comment by portertech** [comment]
You've broken client safe mode, it will never execute checks if true, https://github.com/Igorshp/sensu/blob/check_merge/lib/sensu/client/process.rb#L216

**Comment by Igorshp** [comment]
Restarting Travis Build

File: lib/sensu/client/process.rb [hunk 1]
```diff
@@ -209,11 +209,11 @@ def run_check_extension(check)
...
         @logger.debug("processing check", :check => check)
+        if @settings.check_exists?(check[:name])
+          check.merge!(@settings[:checks][check[:name]])
+        end
         if check.has_key?(:command)
-          if @settings.check_exists?(check[:name])
-            check.merge!(@settings[:checks][check[:name]])
-            execute_check_command(check)
-          elsif @safe_mode
+          if !@settings.check_exists?(check[:name]) && @safe_mode
```
Comment (portertech) [inline lib/sensu/client/process.rb]: let's flip it around, for readability: `@safe_mode && !@settings.check_exists?(check[:name])`

:+1:

**Pull request merged**